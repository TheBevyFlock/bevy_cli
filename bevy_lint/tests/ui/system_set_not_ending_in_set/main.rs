#![feature(register_tool)]
#![register_tool(bevy)]
#![deny(bevy::system_set_not_ending_in_set)]
//~^ NOTE: the lint level is defined here
#![allow(dead_code)]
use bevy::prelude::*;

//~v NOTE: `SystemSet` implemented here
#[derive(SystemSet, Debug, Clone, PartialEq, Eq, Hash)]
struct MyAudio;
//~^ ERROR: implemented `SystemSet` for a struct whose name does not end in "Set"
//~| HELP: rename the SystemSet
