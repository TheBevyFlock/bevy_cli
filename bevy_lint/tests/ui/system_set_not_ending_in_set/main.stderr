error: implemented `SystemSet` for a struct whose name does not end in "Set"
  --> tests/ui/system_set_not_ending_in_set/main.rs:10:8
   |
10 | struct MyAudio;
   |        ^^^^^^^ help: rename the SystemSet: `MyAudioSet`
   |
note: `SystemSet` implemented here
  --> tests/ui/system_set_not_ending_in_set/main.rs:9:10
   |
9  | #[derive(SystemSet, Debug, Clone, PartialEq, Eq, Hash)]
   |          ^^^^^^^^^
note: the lint level is defined here
  --> tests/ui/system_set_not_ending_in_set/main.rs:3:9
   |
3  | #![deny(bevy::system_set_not_ending_in_set)]
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   = note: this error originates in the derive macro `SystemSet` (in Nightly builds, run with -Z macro-backtrace for more info)

error: aborting due to 1 previous error

